.pre-commit-config.yaml

```diff
@@ -145,6 +145,24 @@ repos:
         pass_filenames: false
         require_serial: true
         files: ^.github/workflows/.*$
+      - id: check-log-usage
+        name: Check for proper log usage (use llama_stack.log instead)
+        entry: bash
+        language: system
+        types: [python]
+        pass_filenames: true
+        args:
+          - -c
+          - |
+            matches=$(grep -EnH '^[^#]*\b(import logging|from logging\b)' "$@" | grep -v '# allow-direct-logging' || true)
+            if [ -n "$matches" ]; then
+              # GitHub Actions annotation format
+              while IFS=: read -r file line_num rest; do
+                echo "::error file=$file,line=$line_num::Do not use 'import logging' or 'from logging import' in $file. Use the custom log instead: from llama_stack.log import get_logger; log = get_logger(name=__name__, category=\"core\"). If direct logging is truly needed, add: # allow-direct-logging"
```

Comment:
ensure this says `logger = get_logger()` -- that's the more common pattern in the code.